src/main/kotlin/org/rust/ide/intentions/DemorgansLawIntention.kt

```diff
@@ -0,0 +1,138 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.tree.IElementType
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.parentOfType
+
+class DemorgansLawIntention : PsiElementBaseIntentionAction() {
+    private var m_text = ""
+    override fun getText() = m_text
+    override fun getFamilyName() = text
+    override fun startInWriteAction() = true
+
+    fun getTextForElement(element: PsiElement) {
+        val binaryExpression = element as RustBinaryExprElement
+        m_text = when (binaryExpression.operatorType) {
+            RustTokenElementTypes.ANDAND -> "DeMorgan's Law, Replace '&&' with '||'"
+            RustTokenElementTypes.OROR -> "DeMorgan's Law, Replace '||' with '&&'"
+            else -> ""
+        }
+    }
```

Comment:
You don't need to override `getText`. This `m_text` thing is basically what is already done in the parent class. Instead, you can just assign to `text` directly:

```
text = when ...
```